src/peer.rs

```diff
@@ -48,42 +48,42 @@ impl LachesisPeer for LachesisPeerStruct {
                 id: id,
                 net_addr: net_addr,
                 used: 0,
-                height: -1,
+                height: 0,
                 in_degree: 0,
                 lock: RwLock::new(()),
             }
     }
-    fn set_height(&mut self, height: i64){
+    fn set_height(&mut self, height: u64){
         let _ = self.lock.write().unwrap();
         self.height = height;
     }
-    fn get_height(&self) -> i64 {
+    fn get_height(&self) -> u64 {
         let _ = self.lock.read().unwrap();
         return self.height;
     }
-    fn next_height(&mut self) -> i64 {
+    fn next_height(&mut self) -> u64 {
         let _ = self.lock.write().unwrap();
         self.height += 1;
         return self.height;
     }
-    fn set_in_degree(&mut self, in_degree: i64){
+    fn set_in_degree(&mut self, in_degree: u64){
         let _ = self.lock.write().unwrap();
         self.in_degree = in_degree;
     }
-    fn get_in_degree(&self) -> i64 {
+    fn get_in_degree(&self) -> u64 {
         let _ = self.lock.read().unwrap();
         return self.in_degree;
     }
     fn inc_in_degree(&mut self) {
         let _ = self.lock.write().unwrap();
         self.in_degree += 1;
     }
-    fn get_peers(json_peer_path: String) -> Vec<LachesisPeerStruct> {
-        let mut file = File::open(json_peer_path).expect("get_peers(), open file");
+    fn get_peers_from_file(json_peer_path: String) -> Result<Vec<LachesisPeerStruct>, Error> {
+        let mut file = File::open(json_peer_path)?;
         let mut data = String::new();
-        file.read_to_string(&mut data).expect("reading from json file");
-        let Ar: Vec<LachesisPeerStruct> = serde_json::from_str(&data).expect("desrialisation from json data");
-        return Ar;
+        file.read_to_string(&mut data)?;
+        let ar: Vec<LachesisPeerStruct> = serde_json::from_str(&data)?;
+        Ok(ar)
```

Comment:
It's a matter of preference, but I like to go a little further:
```rust
Ok(serde_json::from_str(&data)?)
```

Reply:
Sure, that is shorter.